src/POData/UriProcessor/ResourcePathProcessor/SegmentParser/TargetKind.php

```diff
@@ -102,4 +106,29 @@ class TargetKind extends Enum
      * A singleton (parameter-less function wrapper).
      */
     protected const SINGLETON = 13;
+
+    /**
+     * Is this segment a terminal segment - nothing else can be added after it?
+     *
+     * @return bool
+     */
+    public function isTerminal(): bool
+    {
+        return array_key_exists($this->getValue(), self::TERMINAL_VALUES);
+    }
+
+    /**
+     * Can this segment be directly processed?
+     *
+     * @return bool
+     */
+    public function isDirectProcess(): bool
```

Comment:
Don't like the name direct process. it appears to encapsulate special purpose paths.
`/` 
`/$metadata` 
and 
`/$batch`

a better name could be picked.